src/main/java/com/shade/decima/ui/data/viewer/model/utils/MathUtils.java

```diff
@@ -0,0 +1,42 @@
+package com.shade.decima.ui.data.viewer.model.utils;
+
+
+public class MathUtils {
+    public static float toFloat(short h) {
+        final int bits = h & 0xffff;
+        final int sign = bits & 0x8000;
+        final int exponent = (bits >>> 10) & 0x1f;
+        final int mantissa = (bits) & 0x3ff;
+
+        int outE = 0;
+        int outM = 0;
+
+        if (exponent == 0) {
+            if (mantissa != 0) {
+                final float o = Float.intBitsToFloat((126 << 23) + mantissa) - Float.intBitsToFloat(126 << 23);
+                return sign == 0 ? o : -o;
+            }
+        } else {
+            outM = mantissa << 13;
+            if (exponent == 0x1f) {
+                outE = 0xff;
+                if (outM != 0) {
+                    outM |= 0x400000;
+                }
+            } else {
+                outE = exponent - 15 + 127;
+            }
+        }
+
+        return Float.intBitsToFloat((sign << 16) | (outE << 23) | outM);
+    }
+
+    public static double dotProduct(double[] vec1, double[] vec2) {
```

Comment:
This method is used in a single place only, please move it and remove this class